cdb2api/cdb2api.c

```diff
@@ -4297,6 +4319,58 @@ static inline void consume_previous_query(cdb2_hndl_tp *hndl)
         goto retry_queries;                                                    \
     } while (0);
 
+// make newly opened child handle have same settings (references) as parent
+static void attach_to_handle(cdb2_hndl_tp *child, cdb2_hndl_tp *parent)
+{
+    child->is_child_hndl = 1;
+    child->bindvars = parent->bindvars;
+    child->n_bindvars = parent->n_bindvars;
+
+    // TODO: Maybe shouldn't have authentication commands in here
+    child->num_set_commands = parent->num_set_commands;
+    child->num_set_commands_sent = 0; // we are opening a new handle (child) so need to send all set commands again
+    child->commands = parent->commands;
+    // process_set_local
+    child->is_hasql = parent->is_hasql;
+    child->is_admin = parent->is_admin;
+
+    // process_ssl_set_command
+    // set ssl vars to parent since process set command might have updated parent handle
+    child->c_sslmode = parent->c_sslmode;
```

Comment:
Can you please remind me again why we need to copy here? Wouldn't rerunning the set commands just populate these variables on the child handle?

Reply:
SSL set commands aren't saved in the handle :(

Reply:
Ah ... that explains it! thank you!

Reply:
uc?